### DISCUSSION TITLE
Veto

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#403] [Solidity / smart contracts] [answered]
Can someone explain to me how the “Veto” works in the noun governance? I’d like to be able implement something like this. I’m working on establishing governance of a treasury via 44 ERC-721s.

**Comment by solimander** [comment, marked as answer]
Veto is an additional state on a proposal that can only be set by a “vetoer” address. It’s a terminal state so once a proposal is vetoed it cant be set to another state. The code for it is mostly taken from the `cancel` function and you can think of it that way: once a proposal has been cancelled or vetoed it can’t be voted on, queued, or executed.
